xen/arch/arm/coproc/coproc.c

```diff
@@ -267,6 +267,22 @@ bool_t coproc_is_attached_to_domain(struct domain *d, const char *path)
     return is_created;
 }
 
+/*
+ * To make a decision can this device be assigned to the IOMMU context.
+ * As we only have device-tree support for now treat potential coproc device
+ * as a device-tree device.
+ */
+bool_t coproc_is_protected(device_t *dev)
```

Comment:
Let me elaborate a bit more about this:
xxx_is_protected means that a device has been protected by IOMMU and can be assigned to domain (passthroughed or shared).
IOMMU device being probed scans for all devices physically connected to it and mark them as protected. Then it run-time, during assigning device to domain (passthrough or shared) IOMMU framework checks if device is protected (in other words -> IOMMU is aware of this device) and allows to assign it to domain if so.